modules/ding_borgun_checkout/ding_borgun_checkout.module

```diff
@@ -0,0 +1,353 @@
+          array(
+            'type' => 'setting',
+            'data' => array('dingBorgunCheckout' => $settings),
+          ),
+        ),
+      ),
+    );
+  }
+}
+
+/**
+ * Form validation handler.
+ */
+function ding_borgun_checkout_validate($form, &$form_state) {
+  $element = $form_state['triggering_element'];
+  $debts = $form_state['build_info']['args'][0];
+
+  $order_id = _ding_borgun_checkout_get_order_id();
+  if (!$order_id) {
+    watchdog('ding_borgun_checkout', 'Could not get lock for order id.', array(), WATCHDOG_ERROR);
+    form_set_error('', t('Error acquiring payment, please try again later.'));
+  }
+
+  $debt_ids = array_map(function ($debt) {
+    return $debt->id;
+  }, $debts);
+  $form_state['debt_ids'] = $debt_ids;
+  $meta = 'Payment for: ' . implode(', ', $debt_ids);
+
+  $form_state['transaction_id'] = _ding_borgun_checkout_capture_token(
+    $form_state['values']['BorgunCheckoutToken'],
+    $element['#amount'],
+    $element['#currency_code'],
+    $order_id,
+    $meta);
+  if (!$form_state['transaction_id']) {
+    form_set_error('', t('Error acquiring payment, please try again later.'));
+  }
+}
+
+/**
+ * Form submit handler.
+ */
+function ding_borgun_checkout_submit($form, &$form_state) {
+  $element = $form_state['triggering_element'];
+  ding_provider_invoke('debt', 'payment_received', $element['#amount'], $form_state['debt_ids'], $form_state['transaction_id']);
+}
+
+/**
+ * Get the host to use.
+ *
+ * Returns the proper hostname for test or live environments.
+ *
+ * @return string
+ *   The base url, without trailing slash.
+ */
+function _ding_borgun_checkout_host() {
+  return variable_get('ding_borgun_checkout_host', 'https://test.borgun.is');
+}
+
+/**
+ * Get an order id.
+ *
+ * Returns an unique order id. Borgun order ids needs to be exactly 12
+ * characters and can contain A-Z0-9.
+ *
+ * @return string|null
+ *   The order id, or null if locking failed.
+ */
+function _ding_borgun_checkout_get_order_id() {
+  $lock_id = 'ding_borgun_checkout_order_id';
+  // Use a lock to ensure that there's only one process that increment the order
+  // id at a time.
+  $locked = lock_acquire($lock_id);
+  if (!$locked) {
+    if (!lock_wait($lock_id, 30)) {
+      $locked = lock_acquire($lock_id);
+    }
+  }
+
+  if (!$locked) {
+    return NULL;
+  }
+
+  $counter = variable_get('ding_borgun_checkout_order_id_next', '1');
+  variable_set('ding_borgun_checkout_order_id_next', $counter + 1);
+  lock_release($lock_id);
+  $prefix = variable_get('ding_borgun_checkout_order_id_prefix', '');
+  // Using strlen is safe, we've limited the prefix to A-Z.
+  $padding = 12 - strlen($prefix);
+  return sprintf('%s%0' . $padding . 'd', $prefix, $counter);
+}
+
+/**
+ * Capture single use token.
+ *
+ * Authorizes and captures a payment using a single use token from
+ * Borgun.Checkout.
+ *
+ * @param string $token
+ *   Token supplied by Borgun.Checkout.
+ * @param int|float $amount
+ *   Payment amount.
```

Comment:
If this is to be an amount with 2 decimal points then that should probably be documented here.